aspnetcore/fundamentals/minimal-apis/includes/parameter-binding8.md

```diff
@@ -63,6 +63,18 @@ An alternative is to use the [`[AsParameters]`](xref:Microsoft.AspNetCore.Http.A
 
 For more information, see the section on [AsParameters](#parameter-binding-for-argument-lists-with-asparameters) later in this article.
 
+The [complete sample code](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/fundamentals/minimal-apis/samples/IFormFile) is in the [AspNetCore.Docs.Samples](https://github.com/dotnet/AspNetCore.Docs.Samples) repository.
+
+#### Secure binding from IFormFile and IFormFileCollection
+
+Complex form binding is supported using <xref:Microsoft.AspNetCore.Http.IFormFile> and <xref:Microsoft.AspNetCore.Http.IFormFileCollection> using the [`[FromForm]`](xref:Microsoft.AspNetCore.Mvc.FromFormAttribute):
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/minimal-apis/samples/IFormFile/Program.cs" id="snippet_1":::
```

Comment:
The sample code shows an example of validating the form manually in an endpoint:

```csharp
app.MapPost("/upload", async Task<Results<Ok<string>,BadRequest<string>>>
    ([FromForm] FileUploadForm fileUploadForm, HttpContext context,
                                                IAntiforgery antiforgery) =>
{
    try
    {
        await antiforgery.ValidateRequestAsync(context);
        await MyUtils.SaveFileWithName(fileUploadForm.FileDocument!,
                  fileUploadForm.Name!, app.Environment.ContentRootPath);
        return TypedResults.Ok($"Your file with the description:" +
            $" {fileUploadForm.Description} has been uploaded successfully");
    }
    catch (AntiforgeryValidationException e)
    {
        return TypedResults.BadRequest("Invalid anti-forgery token" + e.HResult);
    }
});
```

But I don't think that's necessary since preview 7(?). Now the antiforgery-token is automatically validated if you have a `[FromForm]` parameter in your API. So you could simplify this to:

```diff
app.MapPost("/upload", async Task<Results<Ok<string>,BadRequest<string>>>
    ([FromForm] FileUploadForm fileUploadForm, HttpContext context,
                                                IAntiforgery antiforgery) =>
{
-   try
-   {
-       await antiforgery.ValidateRequestAsync(context);
        await MyUtils.SaveFileWithName(fileUploadForm.FileDocument!,
                  fileUploadForm.Name!, app.Environment.ContentRootPath);
        return TypedResults.Ok($"Your file with the description:" +
            $" {fileUploadForm.Description} has been uploaded successfully");
-   }
-   catch (AntiforgeryValidationException e)
-   {
-       return TypedResults.BadRequest("Invalid anti-forgery token" + e.HResult);
-   }
});
```

That said, there _is_ value in showing how to do it manually, but I think this would now mostly be useful for cases where you're _not_ using model-binding, and instead using `Request.ReadFormAsync()` to manually read the form.